addons/sms/models/mail_thread.py

```diff
@@ -368,3 +368,14 @@ def _notify_thread_by_sms(self, message, recipients_data, msg_vals=False,
             sms_all.filtered(lambda sms: sms.state == 'outgoing').send(auto_commit=False, raise_exception=False)
 
         return True
+
+    def _notify_cancel_sms(self):
+        # TDE CHECK: delete pending SMS
+        self._notify_cancel_by_type_std('sms')
+
+    @api.model
+    def notify_cancel_by_type(self, notification_type):
+        super().notify_cancel_by_type(notification_type)
+        if notification_type == 'sms':
+            self._notify_cancel_sms()
```

Comment:
Could probably call directly _notify_cancel_by_type_std ?

Reply:
Changed (also for email)